### ISSUE TITLE
level_insert_ssts debug_assert failed.

### ISSUE THREAD

**Issue by chenzl25** [risingwavelabs/risingwave#6342]
### Describe the bug

```
<unknown> 0x0000000189f7f224
<unknown> 0x0000000189eef2c8
abort_internal mod.rs:317
abort process.rs:2119
{closure#0} lib.rs:88
{closure#0}<risingwave_rt::set_panic_hook::{closure_env#0}> panicking.rs:232
[Inlined] _$LT$alloc..boxed..Box$LT$F$C$A$GT$$u20$as$u20$core..ops..function..Fn$LT$Args$GT$$GT$::call boxed.rs:1952
rust_panic_with_hook panicking.rs:692
{closure#0} panicking.rs:577
__rust_end_short_backtrace<std::panicking::begin_panic_handler::{closure_env#0}, !> backtrace.rs:138
begin_panic_handler panicking.rs:575
panic_fmt panicking.rs:65
panic panicking.rs:115
level_insert_ssts hummock_version_ext.rs:625
apply_compact_ssts hummock_version_ext.rs:503
apply_version_delta hummock_version_ext.rs:372
{async_fn#0}<risingwave_meta::storage::mem_meta_store::MemStore> mod.rs:1073
poll<risingwave_meta::hummock::manager::{impl#0}::report_compact_task_impl::{async_fn_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>> mod.rs:91
{async_fn#0}<risingwave_meta::storage::mem_meta_store::MemStore> mod.rs:739
poll<risingwave_meta::hummock::manager::{impl#0}::get_compact_task_impl::{async_fn_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>> mod.rs:91
{async_fn#0}<risingwave_meta::storage::mem_meta_store::MemStore> mod.rs:836
poll<risingwave_meta::hummock::manager::{impl#0}::get_compact_task::{async_fn_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>> mod.rs:91
{async_fn#0}<risingwave_meta::storage::mem_meta_store::MemStore> compaction_scheduler.rs:237
poll<risingwave_meta::hummock::compaction_scheduler::{impl#1}::pick_and_assign_impl::{async_fn_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>> mod.rs:91
{async_fn#0}<risingwave_meta::storage::mem_meta_store::MemStore> compaction_scheduler.rs:191
poll<risingwave_meta::hummock::compaction_scheduler::{impl#1}::pick_and_assign::{async_fn_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>> mod.rs:91
{async_fn#0}<risingwave_meta::storage::mem_meta_store::MemStore> compaction_scheduler.rs:176
poll<risingwave_meta::hummock::compaction_scheduler::{impl#1}::start::{async_fn_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>> mod.rs:91
{async_block#0}<risingwave_meta::storage::mem_meta_store::MemStore> mod.rs:160
poll<risingwave_meta::hummock::start_compaction_scheduler::{async_block_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>> mod.rs:91
poll<alloc::boxed::Box<core::future::from_generator::GenFuture<risingwave_meta::hummock::start_compaction_scheduler::{async_block_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>>, alloc::alloc::Global>> future.rs:124
poll<core::pin::Pin<alloc::boxed::Box<core::future::from_generator::GenFuture<risingwave_meta::hummock::start_compaction_scheduler::{async_block_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>>, alloc::alloc::Global>>> instrument.rs:272
{closure#0}<tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<core::future::from_generator::GenFuture<risingwave_meta::hummock::start_compaction_scheduler::{async_block_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>>, alloc::alloc::Global>>>> core.rs:184
with_mut<tokio::runtime::task::core::Stage<tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<core::future::from_generator::GenFuture<risingwave_meta::hummock::start_compaction_scheduler::{async_block_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>>, alloc::alloc::Global>>>>, core::task::poll::Poll<()>, tokio::runtime::task::core::{impl#3}::poll::{closure_env#0}<tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<core::future::from_generator::GenFuture<risingwave_meta::hummock::start_compaction_scheduler::{async_block_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>>, alloc::alloc::Global>>>>> unsafe_cell.rs:14
poll<tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<core::future::from_generator::GenFuture<risingwave_meta::hummock::start_compaction_scheduler::{async_block_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>>, alloc::alloc::Global>>>> core.rs:174
{closure#0}<tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<core::future::from_generator::GenFuture<risingwave_meta::hummock::start_compaction_scheduler::{async_block_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>>, alloc::alloc::Global>>>, alloc::sync::Arc<tokio::runtime::scheduler::multi_thread::worker::Shared>> harness.rs:480
call_once<core::task::poll::Poll<()>, tokio::runtime::task::harness::poll_future::{closure_env#0}<tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<core::future::from_generator::GenFuture<risingwave_meta::hummock::start_compaction_scheduler::{async_block_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>>, alloc::alloc::Global>>>, alloc::sync::Arc<tokio::runtime::scheduler::multi_thread::worker::Shared>>> unwind_safe.rs:271
do_call<core::panic::unwind_safe::AssertUnwindSafe<tokio::runtime::task::harness::poll_future::{closure_env#0}<tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<core::future::from_generator::GenFuture<risingwave_meta::hummock::start_compaction_scheduler::{async_block_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>>, alloc::alloc::Global>>>, alloc::sync::Arc<tokio::runtime::scheduler::multi_thread::worker::Shared>>>, core::task::poll::Poll<()>> panicking.rs:483
__rust_try 0x0000000109d76714
try<core::task::poll::Poll<()>, core::panic::unwind_safe::AssertUnwindSafe<tokio::runtime::task::harness::poll_future::{closure_env#0}<tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<core::future::from_generator::GenFuture<risingwave_meta::hummock::start_compaction_scheduler::{async_block_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>>, alloc::alloc::Global>>>, alloc::sync::Arc<tokio::runtime::scheduler::multi_thread::worker::Shared>>>> panicking.rs:447
catch_unwind<core::panic::unwind_safe::AssertUnwindSafe<tokio::runtime::task::harness::poll_future::{closure_env#0}<tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<core::future::from_generator::GenFuture<risingwave_meta::hummock::start_compaction_scheduler::{async_block_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>>, alloc::alloc::Global>>>, alloc::sync::Arc<tokio::runtime::scheduler::multi_thread::worker::Shared>>>, core::task::poll::Poll<()>> panic.rs:137
poll_future<tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<core::future::from_generator::GenFuture<risingwave_meta::hummock::start_compaction_scheduler::{async_block_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>>, alloc::alloc::Global>>>, alloc::sync::Arc<tokio::runtime::scheduler::multi_thread::worker::Shared>> harness.rs:468
poll_inner<tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<core::future::from_generator::GenFuture<risingwave_meta::hummock::start_compaction_scheduler::{async_block_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>>, alloc::alloc::Global>>>, alloc::sync::Arc<tokio::runtime::scheduler::multi_thread::worker::Shared>> harness.rs:104
poll<tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<core::future::from_generator::GenFuture<risingwave_meta::hummock::start_compaction_scheduler::{async_block_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>>, alloc::alloc::Global>>>, alloc::sync::Arc<tokio::runtime::scheduler::multi_thread::worker::Shared>> harness.rs:57
poll<tracing::instrument::Instrumented<core::pin::Pin<alloc::boxed::Box<core::future::from_generator::GenFuture<risingwave_meta::hummock::start_compaction_scheduler::{async_block_env#0}<risingwave_meta::storage::mem_meta_store::MemStore>>, alloc::alloc::Global>>>, alloc::sync::Arc<tokio::runtime::scheduler::multi_thread::worker::Shared>> raw.rs:194
poll raw.rs:134
run<alloc::sync::Arc<tokio::runtime::scheduler::multi_thread::worker::Shared>> mod.rs:385
{closure#0} worker.rs:444
{closure#0}<core::result::Result<alloc::boxed::Box<tokio::runtime::scheduler::multi_thread::worker::Core, alloc::alloc::Global>, ()>, tokio::runtime::scheduler::multi_thread::worker::{impl#1}::run_task::{closure_env#0}> coop.rs:102
try_with<core::cell::Cell<tokio::coop::Budget>, tokio::coop::with_budget::{closure_env#0}<core::result::Result<alloc::boxed::Box<tokio::runtime::scheduler::multi_thread::worker::Core, alloc::alloc::Global>, ()>, tokio::runtime::scheduler::multi_thread::worker::{impl#1}::run_task::{closure_env#0}>, core::result::Result<alloc::boxed::Box<tokio::runtime::scheduler::multi_thread::worker::Core, alloc::alloc::Global>, ()>> local.rs:446
with<core::cell::Cell<tokio::coop::Budget>, tokio::coop::with_budget::{closure_env#0}<core::result::Result<alloc::boxed::Box<tokio::runtime::scheduler::multi_thread::worker::Core, alloc::alloc::Global>, ()>, tokio::runtime::scheduler::multi_thread::worker::{impl#1}::run_task::{closure_env#0}>, core::result::Result<alloc::boxed::Box<tokio::runtime::scheduler::multi_thread::worker::Core, alloc::alloc::Global>, ()>> local.rs:422
[Inlined] tokio::coop::with_budget coop.rs:95
[Inlined] tokio::coop::budget coop.rs:72
run_task worker.rs:420
run worker.rs:387
{closure#0} worker.rs:372
set<tokio::runtime::scheduler::multi_thread::worker::Context, tokio::runtime::scheduler::multi_thread::worker::run::{closure_env#0}, ()> scoped_tls.rs:61
run worker.rs:369
{closure#0} worker.rs:348
poll<tokio::runtime::scheduler::multi_thread::worker::{impl#0}::launch::{closure_env#0}, ()> task.rs:42
poll<tokio::runtime::blocking::task::BlockingTask<tokio::runtime::scheduler::multi_thread::worker::{impl#0}::launch::{closure_env#0}>> instrument.rs:272
{closure#0}<tracing::instrument::Instrumented<tokio::runtime::blocking::task::BlockingTask<tokio::runtime::scheduler::multi_thread::worker::{impl#0}::launch::{closure_env#0}>>> core.rs:184
with_mut<tokio::runtime::task::core::Stage<tracing::instrument::Instrumented<tokio::runtime::blocking::task::BlockingTask<tokio::runtime::scheduler::multi_thread::worker::{impl#0}::launch::{closure_env#0}>>>, core::task::poll::Poll<()>, tokio::runtime::task::core::{impl#3}::poll::{closure_env#0}<tracing::instrument::Instrumented<tokio::runtime::blocking::task::BlockingTask<tokio::runtime::scheduler::multi_thread::worker::{impl#0}::launch::{closure_env#0}>>>> unsafe_cell.rs:14
poll<tracing::instrument::Instrumented<tokio::runtime::blocking::task::BlockingTask<tokio::runtime::scheduler::multi_thread::worker::{impl#0}::launch::{closure_env#0}>>> core.rs:174
{closure#0}<tracing::instrument::Instrumented<tokio::runtime::blocking::task::BlockingTask<tokio::runtime::scheduler::multi_thread::worker::{impl#0}::launch::{closure_env#0}>>, tokio::runtime::blocking::schedule::NoopSchedule> harness.rs:480
call_once<core::task::poll::Poll<()>, tokio::runtime::task::harness::poll_future::{closure_env#0}<tracing::instrument::Instrumented<tokio::runtime::blocking::task::BlockingTask<tokio::runtime::scheduler::multi_thread::worker::{impl#0}::launch::{closure_env#0}>>, tokio::runtime::blocking::schedule::NoopSchedule>> unwind_safe.rs:271
do_call<core::panic::unwind_safe::AssertUnwindSafe<tokio::runtime::task::harness::poll_future::{closure_env#0}<tracing::instrument::Instrumented<tokio::runtime::blocking::task::BlockingTask<tokio::runtime::scheduler::multi_thread::worker::{impl#0}::launch::{closure_env#0}>>, tokio::runtime::blocking::schedule::NoopSchedule>>, core::task::poll::Poll<()>> panicking.rs:483
__rust_try 0x000000010cb812fc
try<core::task::poll::Poll<()>, core::panic::unwind_safe::AssertUnwindSafe<tokio::runtime::task::harness::poll_future::{closure_env#0}<tracing::instrument::Instrumented<tokio::runtime::blocking::task::BlockingTask<tokio::runtime::scheduler::multi_thread::worker::{impl#0}::launch::{closure_env#0}>>, tokio::runtime::blocking::schedule::NoopSchedule>>> panicking.rs:447
catch_unwind<core::panic::unwind_safe::AssertUnwindSafe<tokio::runtime::task::harness::poll_future::{closure_env#0}<tracing::instrument::Instrumented<tokio::runtime::blocking::task::BlockingTask<tokio::runtime::scheduler::multi_thread::worker::{impl#0}::launch::{closure_env#0}>>, tokio::runtime::blocking::schedule::NoopSchedule>>, core::task::poll::Poll<()>> panic.rs:137
poll_future<tracing::instrument::Instrumented<tokio::runtime::blocking::task::BlockingTask<tokio::runtime::scheduler::multi_thread::worker::{impl#0}::launch::{closure_env#0}>>, tokio::runtime::blocking::schedule::NoopSchedule> harness.rs:468
poll_inner<tracing::instrument::Instrumented<tokio::runtime::blocking::task::BlockingTask<tokio::runtime::scheduler::multi_thread::worker::{impl#0}::launch::{closure_env#0}>>, tokio::runtime::blocking::schedule::NoopSchedule> harness.rs:104
poll<tracing::instrument::Instrumented<tokio::runtime::blocking::task::BlockingTask<tokio::runtime::scheduler::multi_thread::worker::{impl#0}::launch::{closure_env#0}>>, tokio::runtime::blocking::schedule::NoopSchedule> harness.rs:57
poll<tracing::instrument::Instrumented<tokio::runtime::blocking::task::BlockingTask<tokio::runtime::scheduler::multi_thread::worker::{impl#0}::launch::{closure_env#0}>>, tokio::runtime::blocking::schedule::NoopSchedule> raw.rs:194
poll raw.rs:134
run<tokio::runtime::blocking::schedule::NoopSchedule> mod.rs:422
run pool.rs:111
run pool.rs:346
{closure#0} pool.rs:321
__rust_begin_short_backtrace<tokio::runtime::blocking::pool::{impl#5}::spawn_thread::{closure_env#0}, ()> backtrace.rs:122
{closure#0}<tokio::runtime::blocking::pool::{impl#5}::spawn_thread::{closure_env#0}, ()> mod.rs:551
call_once<(), std::thread::{impl#0}::spawn_unchecked_::{closure#1}::{closure_env#0}<tokio::runtime::blocking::pool::{impl#5}::spawn_thread::{closure_env#0}, ()>> unwind_safe.rs:271
do_call<core::panic::unwind_safe::AssertUnwindSafe<std::thread::{impl#0}::spawn_unchecked_::{closure#1}::{closure_env#0}<tokio::runtime::blocking::pool::{impl#5}::spawn_thread::{closure_env#0}, ()>>, ()> panicking.rs:483
__rust_try 0x000000010cb812fc
try<(), core::panic::unwind_safe::AssertUnwindSafe<std::thread::{impl#0}::spawn_unchecked_::{closure#1}::{closure_env#0}<tokio::runtime::blocking::pool::{impl#5}::spawn_thread::{closure_env#0}, ()>>> panicking.rs:447
catch_unwind<core::panic::unwind_safe::AssertUnwindSafe<std::thread::{impl#0}::spawn_unchecked_::{closure#1}::{closure_env#0}<tokio::runtime::blocking::pool::{impl#5}::spawn_thread::{closure_env#0}, ()>>, ()> panic.rs:137
{closure#1}<tokio::runtime::blocking::pool::{impl#5}::spawn_thread::{closure_env#0}, ()> mod.rs:550
call_once<std::thread::{impl#0}::spawn_unchecked_::{closure_env#1}<tokio::runtime::blocking::pool::{impl#5}::spawn_thread::{closure_env#0}, ()>, ()> function.rs:251
[Inlined] _$LT$alloc..boxed..Box$LT$F$C$A$GT$$u20$as$u20$core..ops..function..FnOnce$LT$Args$GT$$GT$::call_once boxed.rs:1938
[Inlined] _$LT$alloc..boxed..Box$LT$F$C$A$GT$$u20$as$u20$core..ops..function..FnOnce$LT$Args$GT$$GT$::call_once boxed.rs:1938
thread_start thread.rs:108
<unknown> 0x0000000189fb606c
```

### To Reproduce

1. Change `risingwave.toml`
```
barrier_interval_ms = 10
checkpoint_frequency = 2
```

2. Use sysbench to keep inserting data to table `sbtest1`
```
sysbench \
--db-driver=pgsql \
--pgsql-host=127.0.0.1  \
--pgsql-port=4566  \
--pgsql-user=root \
--pgsql-db=dev \
--threads=16 \
--report-interval=1 \
--events=0 \
--time=10 \
--percentile=99 \
--auto_inc=false \
--table_size=100000000 \
--db-ps-mode=disable \
--skip_trx=true \
--range_size=1 \
--random_points=1 \
/usr/local/share/sysbench/select_random_points.lua \
prepare
```

3. Create a mv (with another terminal so that we can run create mv concurrently with the insert statements generated by the sysbench)
```sql
 create materialized view v as select count(*) c from sbtest1;
```

4. Wait a minute, it will fail.

**Comment by chenzl25** [comment]
cc @wenym1 @hzxa21

**Comment by hzxa21** [comment]
@chenzl25 How to reproduce?

**Comment by chenzl25** [comment]
> @chenzl25 How to reproduce?

Sorry for the lack of reproducing method before. I have added it now.

**Comment by hzxa21** [comment]
It panicked at the `can_concat` check 
```
2022-11-14T07:32:49.574114Z  INFO risingwave_meta::hummock::compactor_manager: Added compactor session 1
thread 'risingwave-main' panicked at 'assertion failed: can_concat(&operand.table_infos.iter().collect_vec())', src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs:625:5
stack backtrace:
   0: rust_begin_unwind
             at /rustc/b8c35ca26b191bb9a9ac669a4b3f4d3d52d97fb1/library/std/src/panicking.rs:575:5
   1: core::panicking::panic_fmt
             at /rustc/b8c35ca26b191bb9a9ac669a4b3f4d3d52d97fb1/library/core/src/panicking.rs:65:14
   2: core::panicking::panic
             at /rustc/b8c35ca26b191bb9a9ac669a4b3f4d3d52d97fb1/library/core/src/panicking.rs:115:5
   3: risingwave_hummock_sdk::compaction_group::hummock_version_ext::level_insert_ssts
             at ./src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs:625:5
   4: <risingwave_pb::hummock::hummock_version::Levels as risingwave_hummock_sdk::compaction_group::hummock_version_ext::HummockLevelsExt>::apply_compact_ssts
             at ./src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs:503:21
   5: <risingwave_pb::hummock::HummockVersion as risingwave_hummock_sdk::compaction_group::hummock_version_ext::HummockVersionExt>::apply_version_delta
             at ./src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs:372:17
   6: risingwave_meta::hummock::manager::HummockManager<S>::report_compact_task_impl::{{closure}}
             at ./src/meta/src/hummock/manager/mod.rs:1118:17
   7: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/b8c35ca26b191bb9a9ac669a4b3f4d3d52d97fb1/library/core/src/future/mod.rs:91:19
   8: risingwave_meta::hummock::manager::HummockManager<S>::get_compact_task_impl::{{closure}}
             at ./src/meta/src/hummock/manager/mod.rs:778:17
   9: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/b8c35ca26b191bb9a9ac669a4b3f4d3d52d97fb1/library/core/src/future/mod.rs:91:19
  10: risingwave_meta::hummock::manager::HummockManager<S>::get_compact_task::{{closure}}
             at ./src/meta/src/hummock/manager/mod.rs:875:13
  11: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/b8c35ca26b191bb9a9ac669a4b3f4d3d52d97fb1/library/core/src/future/mod.rs:91:19
  12: risingwave_meta::hummock::compaction_scheduler::CompactionScheduler<S>::pick_and_assign_impl::{{closure}}
             at ./src/meta/src/hummock/compaction_scheduler.rs:237:13
  13: <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
```

The SSTs are from non-overlapping level but their key ranges overlap:
```
[
    SstableInfo { id: 53, key_range: Some(KeyRange { left: [0, 0, 3, 237, 0, 0, 11, 234, 98, 142, 90, 0, 0], right: [0, 0, 3, 237, 0, 0, 11, 234, 98, 133, 107, 0, 0] }), file_size: 598, table_ids: [1005], meta_offset: 482, stale_key_count: 14, total_key_count: 15, divide_version: 0 }, 
    SstableInfo { id: 35, key_range: Some(KeyRange { left: [0, 0, 3, 237, 0, 0, 11, 234, 98, 131, 241, 0, 0], right: [0, 0, 3, 237, 0, 0, 11, 234, 98, 126, 38, 0, 0] }), file_size: 596, table_ids: [1005], meta_offset: 480, stale_key_count: 14, total_key_count: 15, divide_version: 0 }
]
```

I believe this is also the cause for #6343

**Comment by hzxa21** [comment]
It is caused by L0 -> L0 compaction task trival move:
```
Compaction watermark: 3354041147064320 
Compaction target_file_size: 33554432 
Compaction # splits: 1 
Compaction task status: Success 
Compaction Sstables structure: 
Level 0 ["[id: 44, 0KB]"] 
Level 0 [] 
Compaction task output: 
SstableInfo: id=44, KeyRange=["000003ed00000bea7ba5a70000","000003ed00000bea7b9daa0000"], size=0KB, delete_ratio=92%
```

**Comment by Little-Wallace** [comment]
This bug is caused by we calculate overlap files by `MinOverlappingPicker::pick_tables` but this function always assume that the next level is `NonOverlapping`. So we can not use it between two `Overlapping` sub-levels

**Comment by Little-Wallace** [comment]
> The SSTs are from non-overlapping level but their key ranges overlap:
> 
> ```
> [
>     SstableInfo { id: 53, key_range: Some(KeyRange { left: [0, 0, 3, 237, 0, 0, 11, 234, 98, 142, 90, 0, 0], right: [0, 0, 3, 237, 0, 0, 11, 234, 98, 133, 107, 0, 0] }), file_size: 598, table_ids: [1005], meta_offset: 482, stale_key_count: 14, total_key_count: 15, divide_version: 0 }, 
>     SstableInfo { id: 35, key_range: Some(KeyRange { left: [0, 0, 3, 237, 0, 0, 11, 234, 98, 131, 241, 0, 0], right: [0, 0, 3, 237, 0, 0, 11, 234, 98, 126, 38, 0, 0] }), file_size: 596, table_ids: [1005], meta_offset: 480, stale_key_count: 14, total_key_count: 15, divide_version: 0 }
> ]
> ```
> 
> I believe this is also the cause for #6343

I run code for these two sstableinfos and found that they does not overlap with each other. So this error is just cause by `can_concat`.

```rust
    #[test]
    fn test_cmp_encode_key() {
        let a_left = vec![0, 0, 3, 237, 0, 0, 11, 234, 98, 142, 90, 0, 0];
        let a_right = vec![0, 0, 3, 237, 0, 0, 11, 234, 98, 133, 107, 0, 0];
        let b_left = vec![0, 0, 3, 237, 0, 0, 11, 234, 98, 131, 241, 0, 0];
        let b_right = vec![0, 0, 3, 237, 0, 0, 11, 234, 98, 126, 38, 0, 0];
        assert!(KeyComparator::compare_encoded_full_key(&a_left, &a_right) == Ordering::Less);
        assert!(KeyComparator::compare_encoded_full_key(&a_right, &b_left) == Ordering::Less);
        assert!(KeyComparator::compare_encoded_full_key(&b_left, &b_right) == Ordering::Less);
    }
```

**Comment by hzxa21** [comment]
resolved by #6364
